src/profiler/cpuProfilerModel.ts

```diff
@@ -0,0 +1,374 @@
+      // Assumption: Javascript is single threaded, so there should only be one thread throughout
+      const tid: string = samples[0].tid;
+      // TODO: What role does id play in string parsing
+      const id: string = '0x1';
+      const startTime: number = Number(samples[0].ts);
+      const { nodes, sampleNumbers, timeDeltas } = this.constructNodes(
+        samples,
+        stackFrames
+      );
+      return {
+        id,
+        pid,
+        tid,
+        startTime,
+        nodes,
+        samples: sampleNumbers,
+        timeDeltas,
+      };
+    } else {
+      throw new Error('The hermes profile has zero samples');
+    }
+  }
+
+  /**
+   * Constructs CPUProfileChunk Nodes and the resultant samples and time deltas to be inputted into the
+   * CPUProfileChunk object which will be processed to give createStartEndEvents()
+   *
+   * @param {HermesSample} samples
+   * @param {Map<string, HermesStackFrame>} stackFrames
+   * @return { Array<CPUProfileChunkNode>, Array<number>, Array<number> }
+   */
+  static constructNodes(
+    samples: HermesSample[],
+    stackFrames: { [key in string]: HermesStackFrame }
+  ): CPUProfileChunker {
+    samples = samples.map((sample: HermesSample) => {
+      sample.stackFrameData = stackFrames[sample.sf];
+      return sample;
+    });
+    const stackFrameIds: string[] = Object.keys(stackFrames);
+    // TODO: What should URLs be? Change name to get things from SourceMaps
+
+    const profileNodes: CPUProfileChunkNode[] = stackFrameIds.map(
+      (stackFrameId: string) => {
+        return {
+          id: Number(stackFrameId),
+          callFrame: {
+            line: stackFrames[stackFrameId].line,
+            column: stackFrames[stackFrameId].column,
+            funcLine: stackFrames[stackFrameId].funcLine,
+            funcColumn: stackFrames[stackFrameId].funcColumn,
+            name: stackFrames[stackFrameId].name,
+            category: stackFrames[stackFrameId].category,
+            url: 'TODO',
+          },
+          parent: stackFrames[stackFrameId].parent,
+        };
+      }
+    );
+    const returnedSamples: number[] = [];
+    const timeDeltas: number[] = [];
+    let lastTimeStamp = Number(samples[0].ts);
+    samples.forEach((sample: HermesSample, idx: number) => {
+      returnedSamples.push(sample.sf);
+      if (idx === 0) {
+        timeDeltas.push(0);
+      } else {
+        const timeDiff = Number(sample.ts) - lastTimeStamp;
+        lastTimeStamp = Number(sample.ts);
+        timeDeltas.push(timeDiff);
+      }
+    });
+
+    return {
+      nodes: profileNodes,
+      sampleNumbers: returnedSamples,
+      timeDeltas,
+    };
+  }
+
+  /**
+   * Refer to the source maps for the `index.bundle` file. Throws error if debug server not running
+   * @param {string} metroServerURL
+   * @param {DurationEvent[]} chromeEvents
+   * @throws {Debug Server not running}
+   * @returns {DurationEvent[]}
+   */
+  static async changeNamesToSourceMaps(
```

Comment:
Does this function need to live on this class at all? Its seems unrelated to the main purpose of this class, and doesn't appear to use any internal state?

I'd move it into its own function that gets called after the main profile transformation happens.